src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
+  const tableBody = document.querySelector('tbody');
+
+  tableBody.innerHTML = '';
+
+  for (const item of data) {
+    const row = createRow(item);
+    tableBody.appendChild(row);
+  }
+}
+
+function sortItems(items, targetValue, order) {
+  items.sort((a, b) => {
+    if (targetValue === 'salary') {
+      const aSalary = a[targetValue].slice(1).split(',').join('');
+      const bSalary = b[targetValue].slice(1).split(',').join('');
+
+      return order === ASC ? aSalary - bSalary : bSalary - aSalary;
+    }
+
+    if (targetValue === 'age') {
+      return order === ASC
+        ? a[targetValue] - b[targetValue]
+        : b[targetValue] - a[targetValue];
+    }
+
+    return order === ASC
+      ? a[targetValue].localeCompare(b[targetValue])
+      : b[targetValue].localeCompare(a[targetValue]);
+  });
+}
+
+function createSelectBox(title) {
+  const selectBox = document.createElement('select');
+
+  selectBox.name = title;
+  selectBox.setAttribute('data-qa', title);
+  selectBox.required = true;
+
+  OFFICES.forEach((office) => {
+    const option = document.createElement('option');
+
+    option.value = office;
+    option.innerText = office;
+    selectBox.appendChild(option);
+  });
+
+  return selectBox;
+}
+
+function createInput(title) {
+  const input = document.createElement('input');
+
+  input.name = title;
+  input.required = true;
+
+  if (title === 'age' || title === 'salary') {
+    input.type = 'number';
+    input.min = 0;
+  } else {
+    input.type = 'text';
+  }
+
+  return input;
+}
+
+function createButton() {
+  const button = document.createElement('button');
+
+  button.type = 'submit';
+  button.innerText = 'Save to table';
+
+  return button;
+}
+
+function addForm() {
+  const body = document.body;
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  for (const title of HEADERS) {
+    const label = document.createElement('label');
+
+    label.innerText = `${title[0].toUpperCase() + title.slice(1)}:`;
+
+    if (title === 'office') {
+      label.appendChild(createSelectBox(title));
+      form.appendChild(label);
+      continue;
+    }
+
+    label.appendChild(createInput(title));
+    form.appendChild(label);
+  }
+
+  form.appendChild(createButton());
+  body.appendChild(form);
+}
+
+const pushNotification = (title, description, type) => {
+  const table = document.querySelector('table');
+
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = `notification ${type}`;
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+  table.insertAdjacentElement('afterend', notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, DELAY_NOTIFICATION);
+};
+
+function isValid(name, age) {
+  if (name.length < 4) {
+    pushNotification('Oops!', 'Value has less than 4 letters', 'error');
+    return false;
```

Comment:
Error handling: It would be better to throw an error with a specific message instead of showing a notification. This would allow you to handle the error at a higher level in your code, which could improve its maintainability.